iscsi/iscsi.go

```diff
@@ -415,83 +408,131 @@ func DisconnectVolume(c Connector) error {
+			return nil, err
+		}
+		debug.Printf("mount target is a multipath device")
+		return multipathDevice, nil
+	}
+
+	if len(c.Devices) == 0 {
+		return nil, fmt.Errorf("could not find mount target device: connector does not contain any device")
+	}
+
+	return &c.Devices[0], nil
+}
+
+// GetIscsiDevice get an SCSI device from a device name
+func GetIscsiDevice(deviceName string) (*Device, error) {
+	iscsiDevices, err := GetIscsiDevices([]string{deviceName})
 	if err != nil {
 		return nil, err
 	}
-	return &scsiDevices[0], nil
+	if len(iscsiDevices) == 0 {
+		return nil, fmt.Errorf("device %q not found", deviceName)
+	}
+	return &iscsiDevices[0], nil
 }
 
-// GetScsiDevices get scsi devices from a device names
-func GetScsiDevices(devices ...string) ([]ScsiDevice, error) {
-	debug.Printf("Getting info about SCSI devices %s.\n", devices)
-	
-	var devicePaths []string
-	for _, device := range devices {
-		devicePaths = append(devicePaths, filepath.Join(devPath, device))
+// GetScsiDevices get SCSI devices from device paths
+// It will returns all SCSI devices if no paths are given
+func GetScsiDevices(devicePaths []string) ([]Device, error) {
+	debug.Printf("Getting info about SCSI devices %s.\n", devicePaths)
+
+	out, err := lsblk("-JS", devicePaths)
+	if err != nil {
+		debug.Printf("An error occured while looking info about SCSI devices: %v", err)
+		return nil, err
 	}
 
-	out, err := exec.Command("lsblk", append([]string{"-SJ"}, devicePaths...)...).Output()
+	var deviceInfo deviceInfo
+	err = json.Unmarshal(out, &deviceInfo)
 	if err != nil {
-		debug.Printf("An error occured while looking info about SCSI devices: %v\n", err)
 		return nil, err
 	}
 
-	var scsiInfo scsiInfo
-	json.Unmarshal(out, &scsiInfo)
-	return scsiInfo.BlockDevices, nil
+	return deviceInfo.BlockDevices, nil
 }
 
-func writeInScsiDeviceFile(hctl string, file string, content string) (bool, error) {
-	filename := filepath.Join(sysScsiPath, hctl, "device", file)
-	debug.Printf("Write %q in %q.\n", content, filename)
-	if f, err := os.OpenFile(filename, os.O_TRUNC|os.O_WRONLY, 0200); err != nil {
-		if os.IsNotExist(err) {
-			return true, nil
-		} else {
-			debug.Printf("Error while opening file %v: %v\n", filename, err)
-			return false, err
-		}
-	} else {
-		defer f.Close()
-		if _, err := f.WriteString(content); err != nil {
-			debug.Printf("Error while writing to file %v: %v", filename, err)
-			return false, err
+// GetIscsiDevices get iSCSI devices from device paths
+// It will returns all iSCSI devices if no paths are given
+func GetIscsiDevices(devicePaths []string) (devices []Device, err error) {
+	scsiDevices, err := GetScsiDevices(devicePaths)
+	if err != nil {
+		return
+	}
+
+	for i := range scsiDevices {
+		device := &scsiDevices[i]
+		if device.Transport == "iscsi" {
+			devices = append(devices, *device)
 		}
 	}
-	return false, nil
+
+	return
+}
+
+func lsblk(flags string, devicePaths []string) ([]byte, error) {
```

Comment:
may be `getBlockDeviceList() or something of that sort function name better?..

Reply:
I think `lsblk` stands better, since the function only wrap the lsblk binary, takes flags as parameters and returns an array of bytes. Calling it `lsblk` make it clear what parameters are expected and what the output means. Calling it `getBlockDeviceList` would imply that the function will return an array of devices, which is not the case.

Reply:
> I think `lsblk` stands better, since the function only wrap the lsblk binary, takes flags as parameters and returns an array of bytes. Calling it `lsblk` make it clear what parameters are expected and what the output means. Calling it `getBlockDeviceList` would imply that the function will return an array of devices, which is not the case.

sure, I am fine with current version too.